src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,704 @@
+                if True then enable_sentiment_analysis = True
+                else disable_sentiment_analysis = False
+            agent_id: The formatted CX Agent ID
+        Returns:
+            a list of the updated test cases
+        """
+        test_cases_list = self._global_test_cases.list_test_cases(
+            agent_id=self.agent_id, include_conversation_turns=True)
+        updated_test_cases_list = []
+        for test_case in test_cases_list:
+            if test_case.name in test_case_ids:
+                test_case.test_case_conversation_turns = (
+                    self._toggle_sentiment_in_convturns(
+                        conv_turns=test_case.test_case_conversation_turns,
+                        sentiment=sentiment)
+                )
+                updated_test_cases_list.append(test_case)
+
+        return updated_test_cases_list
+
+    def _toggle_webhooks_in_convturns(
+        self,
+        conv_turns: List[types.ConversationTurn],
+        is_webhook_enabled: bool) -> List[types.ConversationTurn]:
+        """Sets the webhooks equal to True/False for all conversation turns.
+        Args:
+            conversation_turns: types.ConversationTurn
+        Returns:
+            conversation turns
+        """
+
+        new_conversation_turns = []
+        for conversation_turn in conv_turns:
+            conversation_turn.user_input.is_webhook_enabled = is_webhook_enabled
+            new_conversation_turns.append(conversation_turn)
+
+        return new_conversation_turns
+
+    def toggle_webhooks_test_cases(
+        self,
+        test_case_ids: List[str],
+        is_webhook_enabled: bool) -> List[types.TestCase]:
+        """
+        Enable/disable webhooks for the given test case ids.
+        Args:
+            test_cases: a list of the test case ids
+            is_webhook_enabled: if enable then True else False
+            agent_id: The formatted CX Agent ID
+        Returns:
+            list of the updated test caess
+        """
+        test_cases_list = self._global_test_cases.list_test_cases(
+            agent_id=self.agent_id, include_conversation_turns=True)
+        updated_test_cases_list = []
+        for test_case in test_cases_list:
+            if test_case.name in test_case_ids:
+                test_case.test_case_conversation_turns = (
+                    self._toggle_webhooks_in_convturns(
+                        conv_turns=test_case.test_case_conversation_turns,
+                        is_webhook_enabled=is_webhook_enabled)
+                    )
+                updated_test_cases_list.append(test_case)
+
+        return updated_test_cases_list
+
+    @staticmethod
+    def _reverse_dict(dict_):
+        """Utility method that reverse the dict keys and values
+            Args:
+                dict_: a dictionary
+            Returns:
+                a reversed dictionary
+        """
+
+        return {v: k for k, v in dict_.items()}
+
+    def _get_commons_config(
```

Comment:
@jkshj21 let's update the name of this method to  `_get_resource_map`
The use of `common` is vague and we want to be a bit more semantically meaningful.

Reply:
fixed